src/parser.js

```diff
@@ -84,6 +106,32 @@ function sliceCode(code, start, end) {
     return splitted.slice(start - 1, end).join('\n');
 }
 
+export function markersSliceCode( code, markername )
+{
+    if( markername === '' ) {
+        return code;
+    }
+    else {
+        // various language comment
+        var commentOpen="(\/+\/+|#|%|\/\\*)";
+        var commentClose="(\\*\/)?";
+        var balise="\\["+markername+"\\]";
+        var pattern=commentOpen + ".*\\s*" + balise + ".*\\s*" + commentClose;
+        var regstr=pattern+"([\\s\\S]*)"+pattern;
+        var reg = new RegExp(regstr);
+
+        //var reg = /\[toto\]([\s\S]*)\[toto\]/;
+        var res = code.match(reg);
+        if(res) {
+            return res[3];
+        }
+        else {
+            console.warn('marker `'+markername+'` not found');
+            return [];
```

Comment:
It should be `throw new Error('marker `'+markername+'` not found')`.

Tips: it is smart that use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals "Template literals").

```js
throw new Error(`marker ${markername} not found`);
```

OR 

return warning message(it will be shown in gitbook)

```js
console.warn('marker `'+markername+'` not found');
return 'marker `'+markername+'` not found'
```